sdk/core/core-auth/src/azureSasCredential.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Represents a credential defined by a static shared access signature.
+ */
+export interface SasCredential {
+  /**
+   * The value of the shared access signature represented as a string
+   */
+  readonly signature: string;
+}
+
+/**
+ * A static-signature-based credential that supports updating
+ * the underlying signature value.
+ */
+export class AzureSasCredential implements SasCredential {
+  private _signature: string;
+
+  /**
+   * The value of the signature to be used in authentication
+   */
+  public get signature(): string {
+    return this._signature;
+  }
+
+  /**
+   * Create an instance of an AzureSasCredential for use
+   * with a service client.
+   *
+   * @param signature - The initial value of the signature to use in authentication
+   */
+  constructor(signature: string) {
+    if (!signature) {
+      throw new Error("signature must be a non-empty string");
+    }
+
+    this._signature = signature;
+  }
+
+  /**
+   * Change the value of the signature.
+   *
+   * Updates will take effect upon the next request after
+   * updating the signature value.
+   *
+   * @param newSignature - The new signature value to be used
+   */
+  public update(newSignature: string): void {
+    this._signature = newSignature;
```

Comment:
What's the point in all this indirection? We're not even throwing events or doing anything for observability of update here. We might as well make `signature` a public property and let someone set it directly.

Reply:
Yeah, this is a good point.  I actually think that we should be doing the same thing we do in the constructor and validating that a falsey value isn't passed in.

Reply:
Added some validation here to prevent folks from passing in `null`, `undefined` or `""` here, per our discussion elsewhere in there review.